lib/core/src/persist/cache.rs

```diff
@@ -6,6 +6,7 @@ use super::Persister;
 const KEY_SWAPPER_PROXY_URL: &str = "swapper_proxy_url";
 const KEY_IS_FIRST_SYNC_COMPLETE: &str = "is_first_sync_complete";
 const KEY_WEBHOOK_URL: &str = "webhook_url";
+const KEY_LAST_DERIVATION_INDEX: &str = "last_derivation_index";
```

Comment:
If the cache is wiped or lost, it's reconstructed on next startup without affecting any functionality.

This new key IMO cannot survive this. Once set, it has to be remembered. I suppose the plan is real-time sync will take care of this. If so, maybe add a TODO on this key indicating this shoud be covered by real-time sync?

Reply:
Yeah, but the whole pool is lost also (should also be synced). This will be (re)set to the next unused derivation index the next time an address from the wallet is used.

I added to todo as a reminder [70bf17b](https://github.com/breez/breez-sdk-liquid/pull/531/commits/70bf17bd8ee3833969f07da8a28df8c0fb307419)